lightning/src/ln/channel.rs

```diff
@@ -1656,6 +1657,54 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
 		cmp::min(self.value_to_self_msat as i64 - self.get_outbound_pending_htlc_stats().1 as i64, 0) as u64)
 	}
 
+	fn commit_tx_fee_outbound_htlc(&self) -> u64 {
+		if !self.channel_outbound {
+			return 0;
+		} else {
+			let mut their_acked_htlcs = self.pending_inbound_htlcs.len() as u64;
+
+			for ref htlc in self.pending_outbound_htlcs.iter() {
+				match htlc.state {
+					OutboundHTLCState::Committed => their_acked_htlcs +=1,
+					OutboundHTLCState::RemoteRemoved {..} => their_acked_htlcs +=1,
+					OutboundHTLCState::LocalAnnounced {..} => their_acked_htlcs += 1,
+					_ => {},
+				}
+			}
+
+			for ref htlc in self.holding_cell_htlc_updates.iter() {
+				match htlc {
+					&&HTLCUpdateAwaitingACK::AddHTLC { .. } => their_acked_htlcs += 1,
```

Comment:
6ca654a

Is this true, if we're called through send_htlc_and_commit, we won't flush outgoing HTLC buffer ? We may artificially decrease our value availability and that way fail to send a HTLC, whereas doing one by one would work ?

Reply:
"flush outgoing HTLC buffer" == call `free_holding_cell_htlcs`? 

I thought we can only do that upon receipt of a RAA, since if there's stuff in the holding cell we must be in the state `AwaitingRemoteRevoke`?